File: Lesson1.java
public class Lesson1 {
    public int x;

    public static void DoSomethingElse() {
        System.out.println("DoSomethingElse");
    }

    public void DoSomething(int value) {
        x = value;
        System.out.println("x = " + x);
        DoSomethingElse();
    }

    public static void main(String[] args) {
        Lesson1 obj = new Lesson1();
        obj.DoSomething();
    }
}

File: Lesson1.rb
#!/usr/bin/env ruby
# frozen_string_literals: true

class Lesson1
    @@x = 0
    def initialize(x = 0)
        @x = x
    end

    def DoSomethingElse
        puts 'DoSomethingElse'
    end

    def DoSomething(value)
        @x = value
        puts "x is equal to #{@x}"
        DoSomethingElse()
    end
end

obj = Lesson1.new
obj.DoSomething 4
